### PR TITLE
[Permissions] add user permissions info ui

### PR DESCRIPTION
**Pull request by thenamankumar**
closes: https://github.com/sourcegraph/sourcegraph/issues/47870
closes: https://github.com/sourcegraph/sourcegraph/issues/47878

- Adds accessible repos section under the user permissions page with pagination & search
- Adds View Permissions link in users management actions

https://www.loom.com/share/4c1d2024134741c4802b4d8c43662dbc

## Test plan

Manually tested.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.01% (+0.20 kb) | 0.05% (+7.52 kb) | 0.06% (+7.31 kb) | 0.40% (+3) 🔺 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/naman/user-perms-info-ui/compare-report.html;tab=live_object) for a full comparison between the commits 4a51429 and e06ddae or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx [hunk 1]
```diff
@@ -17,13 +39,44 @@ import styles from './UserSettingsPermissionsPage.module.scss'
...
     React.PropsWithChildren<{
-        user: UserSettingsAreaUserFields
+        user: { id: string; username: string }
     }>
 > = ({ user }) => {
-    useEffect(() => eventLogger.logViewEvent('UserSettingsPermissions'))
-    const permissionsInfo = useObservable(useMemo(() => userPermissionsInfo(user.id), [user.id]))
+    useEffect(() => eventLogger.logViewEvent('UserSettingsPermissions'), [])
+
+    const [{ q }, setSearchQuery] = useURLSyncedState({ q: '' })
```
Comment (sashaostrikov) [inline client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx:47]: I can see that we use `query` usually

```suggestion
    const [{ query }, setSearchQuery] = useURLSyncedState({ query: '' })
```

File: client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx [hunk 2]
```diff
@@ -17,13 +39,44 @@ import styles from './UserSettingsPermissionsPage.module.scss'
...
+    const debouncedQ = useDebounce(q, 300)
```
Comment (sashaostrikov) [inline client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx:48]: full var names fan here 👋🏻 
```suggestion
    const debouncedQuery = useDebounce(query, 300)
```

File: client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx [hunk 3]
```diff
@@ -45,49 +98,108 @@ export const UserSettingsPermissionsPage: React.FunctionComponent<
...
+                <div key={repository.id} className="py-2">
+                    <ExternalRepositoryIcon externalRepo={repository.externalRepository} />
+                    <RepoLink repoName={repository.name} to={repository.url} />
+                </div>
+            ),
+    },
+    {
+        key: 'reason',
+        header: { label: 'Reason', align: 'center' },
+        align: 'center',
```
Comment (sashaostrikov) [inline client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx:173]: it looks better IMO
```suggestion
        align: 'left',
```

File: client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx [hunk 4]
```diff
@@ -45,49 +98,108 @@ export const UserSettingsPermissionsPage: React.FunctionComponent<
...
+        render: ({ reason }: INode) => (
+            <div className="d-flex justify-content-center">
+                <Badge {...(PermissionReasonBadgeProps[reason] || {})}>{reason}</Badge>
+            </div>
+        ),
+    },
+    {
+        key: 'updatedAt',
+        header: { label: 'Updated At', align: 'center' },
+        align: 'center',
```
Comment (sashaostrikov) [inline client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx:183]: it looks better IMO
```suggestion
        align: 'left',
```

File: client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx [hunk 5]
```diff
@@ -45,49 +98,108 @@ export const UserSettingsPermissionsPage: React.FunctionComponent<
...
+            repository && (
+                <div key={repository.id} className="py-2">
+                    <ExternalRepositoryIcon externalRepo={repository.externalRepository} />
+                    <RepoLink repoName={repository.name} to={repository.url} />
+                </div>
+            ),
+    },
+    {
+        key: 'reason',
+        header: { label: 'Reason', align: 'center' },
```
Comment (sashaostrikov) [inline client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx:172]:
```suggestion
        header: { label: 'Reason', align: 'left' },
```

File: client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx [hunk 6]
```diff
@@ -45,49 +98,108 @@ export const UserSettingsPermissionsPage: React.FunctionComponent<
...
+        align: 'center',
+        render: ({ updatedAt }: INode) => (
+            <div className={classNames('d-flex justify-content-center', styles.updatedAtCell)}>
+                {updatedAt ? <Timestamp date={updatedAt} /> : '-'}
+            </div>
+        ),
+    },
+]
+
+const PermissionReasonBadgeProps: { [k: string]: BadgeProps } = {
```
Comment (sashaostrikov) [inline client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx:192]: what does "k" here stand for?

**Review by sashaostrikov — APPROVED** [review]
LGTM! :shipit: 
Added some comments about alignment and var names, nothing much apart from that.

File: client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx (hunk 6 above)
Comment (thenamankumar) [reply client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx:192]: renamed to reason.

File: client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx (hunk 2 above)
Comment (thenamankumar) [reply client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx]: updated

File: client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx (hunk 1 above)
Comment (thenamankumar) [reply client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx]: updated

**Comment by unknwon** [comment]
CleanShot 2023-03-20 at 17 31 24@2x
😍

**Pull request merged**